vm-validator/src/vm_validator.rs

```diff
@@ -79,15 +79,10 @@ impl TransactionValidation for VMValidator {
     }
 
     fn restart(&mut self, config: OnChainConfigPayload) -> Result<()> {
-        let gas_schedule = self.vm.get_gas_schedule()?;
-        let publishing_options = config.get::<VMPublishingOption>()?;
+        let vm_config = config.get::<VMConfig>()?;
         let version = config.get::<LibraVersion>()?;
-        let vm_config = VMConfig {
-            publishing_options,
-            version,
-        };
 
-        self.vm = LibraVM::init_with_config(gas_schedule.clone(), vm_config);
+        self.vm = LibraVM::init_with_config(version, vm_config);
```

Comment:
do we foresee any more configs needed by LibraVM that is not included in the VM Config? (like LibraVersion)

Reply:
I don't think we should have cases like that! My expectation is that `LibraVersion` will determine the code branch whereas the `VMConfig` will carry all information we need. In case we need more data we should version the `VMConfig` struct to contain more fields.